cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserDAO.java

```diff
@@ -525,61 +610,82 @@ private User getUser(final Principal userID, final String usersDN)
-        }
+            String name;
+            if (userID instanceof NumericPrincipal)
+            {
+                name = String.valueOf(uuid2long(UUID.fromString(userID.getName())));
+            }
+            else
+            {
+                name = userID.getName();
+            }
+            Filter notFilter = Filter.createNOTFilter(Filter.createPresenceFilter(LDAP_NSACCOUNTLOCK));
+            Filter equalsFilter = Filter.createEqualityFilter(searchField, name);
+            Filter filter = Filter.createANDFilter(notFilter, equalsFilter);
+            logger.debug("getUser: search filter = " + filter);
 
-        InternalID internalID = getInternalID(uid);
-        ObjectUtil.setField(user, internalID, USER_ID);
-        user.getIdentities().add(new NumericPrincipal(internalID.getUUID()));
+            SearchRequest searchRequest = new SearchRequest(usersDN, SearchScope.ONE, filter, userAttribs);
 
-        String x500str = searchResult.getAttributeValue(userLdapAttrib.get(X500Principal.class));
-        logger.debug("getUser: x500principal = " + x500str);
-        if (x500str != null)
-        {
-            user.getIdentities().add(new X500Principal(x500str));
-        }
+            // Get the search, then loop through the entries & build users
+            // for the list that will return.
+            SearchResult multiSearchResult = getReadOnlyConnection().search(searchRequest);
 
-        String firstName = searchResult.getAttributeValue(LDAP_FIRST_NAME);
-        String lastName = searchResult.getAttributeValue(LDAP_LAST_NAME);
-        if (StringUtil.hasLength(firstName) && StringUtil.hasLength(lastName))
+            if (multiSearchResult == null)
+            {
+                String msg = "getUser: user " + userID.toString() + " not found in " + usersDN;
+                logger.debug(msg);
+                throw new UserNotFoundException(msg);
+            }
+
+            for (SearchResultEntry next : multiSearchResult.getSearchEntries())
+            {
+                userList.add(makeUserFromResult(next));
+            }
+
+        }
+        catch (LDAPException e)
         {
-            user.personalDetails = new PersonalDetails(firstName, lastName);
-            user.personalDetails.address = searchResult.getAttributeValue(LDAP_ADDRESS);
-            user.personalDetails.city = searchResult.getAttributeValue(LDAP_CITY);
-            user.personalDetails.country = searchResult.getAttributeValue(LDAP_COUNTRY);
-            user.personalDetails.email = searchResult.getAttributeValue(LDAP_EMAIL);
-            user.personalDetails.institute = searchResult.getAttributeValue(LDAP_INSTITUTE);
+            LdapDAO.checkLdapResult(e.getResultCode());
         }
 
-        logger.debug("getUser: found " + userID.getName() + " in " + usersDN);
-        return user;
+        logger.debug("getUserList returning " + userList.size() + " entries for " + userID.toString());
```

Comment:
Is this the only place where multiple entries are logged? If so, shouldn't it be a higher level of debugging?

Reply:
Another good point. This particular function is part of the test code (and might get percolated to the cadc-access-control-admin command line interface at a later step.) It's not the only one that will detect multiple ldap entries. There isn't an explicit message blurted if more than one is found (unless it's flagged as an error.) - I think that's a ball drop on my part.  For the sake of tracking the error, I think an info level would be most likely to be seen. ?